subsys/nrf_security/src/drivers/cracen/cracenpsa/src/cracen_mac_cmac.c

```diff
@@ -0,0 +1,186 @@
+		input += operation->bytes_left_for_next_block;
+		input_length -= operation->bytes_left_for_next_block;
+		operation->bytes_left_for_next_block = 0;
+	}
+
+	/* Process the maximum number of full AES blocks. The processing of the
+	 * last byte must be postponed until the operation is finished in @ref
+	 * cracen_cmac_finish. This is because we need to feed the HW some data
+	 * when generating the mac.
+	 */
+	block_bytes = ROUND_DOWN(input_length - 1, AES_BLOCK_SIZE);
+	remaining_bytes = input_length - block_bytes;
+
+	if (operation->bytes_left_for_next_block == 0) {
+		status = sx_mac_feed(&operation->cmac.ctx, operation->input_buffer, AES_BLOCK_SIZE);
+		if (status) {
+			return silex_statuscodes_to_psa(status);
+		}
+		/* Input buffer was processed, reset the bytes for the next block and
+		 * empty input buffer
+		 */
+		operation->bytes_left_for_next_block = AES_BLOCK_SIZE;
+		operation->cmac.is_first_block = false;
+	}
+
+	if (block_bytes) {
+		status = sx_mac_feed(&operation->cmac.ctx, input, block_bytes);
+		if (status) {
+			return silex_statuscodes_to_psa(status);
+		}
+		operation->cmac.is_first_block = false;
+	}
+
+	if (!operation->cmac.is_first_block) {
+		/* save state and wait until processed */
+		status = sx_mac_save_state(&operation->cmac.ctx);
+		if (status) {
+			return silex_statuscodes_to_psa(status);
+		}
+		status = sx_mac_wait(&operation->cmac.ctx);
+		if (status) {
+			return silex_statuscodes_to_psa(status);
+		}
+
+		/* Clear the input_buffer only after the sx_mac_wait call is executed
+		 * since we need to make sure that the DMA transaction is finished
+		 */
+		safe_memzero(operation->input_buffer, sizeof(operation->input_buffer));
+	}
+
+	/* Copy the remaining bytes to the local input buffer */
+	if (remaining_bytes > 0) {
+		memcpy(operation->input_buffer, input + block_bytes, remaining_bytes);
+		operation->bytes_left_for_next_block -= remaining_bytes;
+	}
+
+	return PSA_SUCCESS;
+}
+
+psa_status_t cracen_cmac_finish(cracen_mac_operation_t *operation)
+{
+	int status;
+
+	if (operation->mac_size > AES_BLOCK_SIZE) {
+		return PSA_ERROR_INVALID_ARGUMENT;
+	}
+
+	if (!operation->cmac.is_first_block) {
+		status = sx_mac_resume_state(&operation->cmac.ctx);
+		if (status) {
+			return silex_statuscodes_to_psa(status);
+		}
+	}
+
+	/* If there are data left in the input buffer feed them to the driver.
+	 */
+	if (operation->bytes_left_for_next_block != AES_BLOCK_SIZE) {
+		status = sx_mac_feed(&operation->cmac.ctx, operation->input_buffer,
+				     AES_BLOCK_SIZE - operation->bytes_left_for_next_block);
+		if (status) {
+			return silex_statuscodes_to_psa(status);
+		}
+	}
+
+	/* Generate the MAC. */
+	status = sx_mac_generate(&operation->cmac.ctx, operation->input_buffer);
+	if (status) {
+		return silex_statuscodes_to_psa(status);
+	}
+	status = sx_mac_wait(&operation->cmac.ctx);
+	if (status) {
+		return silex_statuscodes_to_psa(status);
+	}
+
+	return PSA_SUCCESS;
```

Comment:
Could just return `silex_statuscodes_to_psa(status)` directly

Reply:
You are right, will do